driver-core/src/main/com/mongodb/connection/ScramShaAuthenticator.java

```diff
@@ -0,0 +1,437 @@
+/*
```

Comment:
I'd like this broken up into four commits so it's crystal clear what changes were needed for each goal.

1. Just the file rename
2. Any refactoring needed just for clarity
3. SHA-256 support
3. Client key caching

Especially important as this is security-sensitive code.

Reply:
I don't see the need for such micro managing of commit logs, its thankless work for little reward and has a large cost.

I'm happy to split out:
1) SHA-256 support
2) Client Key caching

Reply:
Sorry, I should have explained my intent rather than being so prescriptive.  When reviewing the PR I was obviously most interested in this file, but found it difficult to review because git did not track it as a move but rather a delete of ScramSha1Authenticator.java and an add of ScramShaAuthenticator.java.  So to review the changes to this file I had to resort to manual diffing of the deleted and added files without any help from git. Others who are interested in this change (e.g TSEs and security-sensitive customers who asked for SCRAM-SHA-256) will have the same problem and I'd like to make it as easy as we can for them.

Whatever you think is best to accomplish that goal is fine with me.

Reply:
Closing this PR - as will push a new on with the new history.